### PR TITLE
fix(cmake): cleanup

### PR DESCRIPTION
**Pull request by vomikan**
- Help files replaced;
- bakefiles moved;
- Alpha, Beta, RC variables set;
- gitignore updates.

---
This change is [Reviewable](https://reviewable.io/reviews/moneymanagerex/moneymanagerex/1136)

### CODE REVIEW

File: bootstrap.bat [hunk 1]
```diff
@@ -1,6 +0,0 @@
-@echo off
```
Comment (tomasfrtala) [inline bootstrap.bat]: I do not know where to put this question, so excuse me please. With the cleaning from old building process scripts like `bakefile` and others is it also desired to delete `configure` and `configure.ac` and `Makefile.in`?

File: update_cmake_build.bat [hunk 2]
```diff
@@ -7,7 +7,7 @@ REM          Allows MMEX to be run successfilly within MSVC 2015 IDE
 REM --------------------------------------------------------------------------
```
Comment (tomasfrtala) [inline update_cmake_build.bat]: - Should be this file `update_cmake_build.bat` moved in the util folder `util/update_cmake_build.bat` where are other useful scripts located?
- One naming question. From the content of `update_cmake_build.bat` script the intent is to prepare project files for Microsoft Visual Studio. Therefore the script should be renamed to something like `update_cmake_build_for_msv.bat`?
- The same question applies also to script `update-po-files.sh` to move it to `util/update-po-files.sh`

File: update_cmake_build.bat [hunk 3]
```diff
@@ -31,12 +31,7 @@ copy ".\readme.*"  %mmex_build_dir%
 @echo Updating Help files for: %current_build_type%
 @echo ------------------------------------------------------------------------
 if not exist %mmex_build_dir%\help mkdir %mmex_build_dir%\help
-copy ".\doc\help\*.*"                   "%mmex_build_dir%\help"
-@echo.
-for /f %%d IN ('dir /A:D /B .\doc\help') do (
-    if not exist "%mmex_build_dir%\help\%%d" mkdir "%mmex_build_dir%\help\%%d"
-    copy ".\doc\help\%%d\*.*"  "%mmex_build_dir%\help\%%d"
-)
+xcopy /Y /I /S ".\docs\*.*"                   "%mmex_build_dir%\help"
```
Comment (tomasfrtala) [inline update_cmake_build.bat]: According to my [proposal](https://github.com/moneymanagerex/moneymanagerex/pull/1136/files#diff-af3b638bc2a3e6c650974192a53c7291R271) to rename dir `docs` to `help`, this line needs to be altered too.

File: build/msw-vc-2017/copy_files.bat [hunk 4]
```diff
@@ -6,5 +6,5 @@ copy /Y ..\..\3rd\sorttable.js\sorttable.js %1res
 IF NOT EXIST %1po mkdir %1po
 xcopy /Y /E /I ..\..\po\*.mo %1po\en
 copy /Y ..\..\doc\contrib.txt %1
-IF NOT EXIST %1help xcopy /Y /E /I ..\..\doc\help\*.* %1help
+IF NOT EXIST %1help xcopy /Y /E /I ..\..\docs %1help
```
Comment (tomasfrtala) [inline build/msw-vc-2017/copy_files.bat]: According to my [proposal](https://github.com/moneymanagerex/moneymanagerex/pull/1136/files#diff-af3b638bc2a3e6c650974192a53c7291R271) to rename dir `docs` to `help`, this line needs to be altered too.

File: build/msw-vc-2015/copy_files.bat [hunk 5]
```diff
@@ -6,5 +6,5 @@ copy /Y ..\..\3rd\sorttable.js\sorttable.js %1res
 IF NOT EXIST %1po mkdir %1po
 xcopy /Y /E /I ..\..\po\*.mo %1po\en
 copy /Y ..\..\doc\contrib.txt %1
-IF NOT EXIST %1help xcopy /Y /E /I ..\..\doc\help\*.* %1help
+IF NOT EXIST %1help xcopy /Y /E /I ..\..\docs %1help
```
Comment (tomasfrtala) [inline build/msw-vc-2015/copy_files.bat]: According to my [proposal](https://github.com/moneymanagerex/moneymanagerex/pull/1136/files#diff-af3b638bc2a3e6c650974192a53c7291R271) to rename dir `docs` to `help`, this line needs to be altered too.

File: build/msw-vc-2013e/copy_files.bat [hunk 6]
```diff
@@ -6,5 +6,5 @@ copy /Y ..\..\3rd\sorttable.js\sorttable.js %1res
 IF NOT EXIST %1po mkdir %1po
 xcopy /Y /E /I ..\..\po\*.mo %1po\en
 copy /Y ..\..\doc\contrib.txt %1
-IF NOT EXIST %1help xcopy /Y /E /I ..\..\doc\help\*.* %1help
+IF NOT EXIST %1help xcopy /Y /E /I ..\..\docs %1help
```
Comment (tomasfrtala) [inline build/msw-vc-2013e/copy_files.bat]: According to my [proposal](https://github.com/moneymanagerex/moneymanagerex/pull/1136/files#diff-af3b638bc2a3e6c650974192a53c7291R271) to rename dir `docs` to `help`, this line needs to be altered too.

File: CMakeLists.txt [hunk 7]
```diff
@@ -29,8 +29,11 @@ SET(PROJECT_TARGET_BIN "mmex")
 # ./resources/mmex.rc
 # ---------------------------------------------------------------------------
 SET(MoneyManagerEx_VERSION_MAJOR 1)
-SET(MoneyManagerEx_VERSION_MINOR 3)
-SET(MoneyManagerEx_VERSION_PATCH 3)
+SET(MoneyManagerEx_VERSION_MINOR 4)
+SET(MoneyManagerEx_VERSION_PATCH 0)
+SET(MoneyManagerEx_VERSION_ALPHA 0)
```
Comment (vomikan) [inline CMakeLists.txt]: Yes and yes

File: CMakeLists.txt [hunk 8]
```diff
@@ -265,8 +268,8 @@ INSTALL(FILES
         "${CMAKE_SOURCE_DIR}/README.TXT"
         DESTINATION "${INSTALL_DOC_DIR}")
 INSTALL(DIRECTORY
-        "${CMAKE_SOURCE_DIR}/doc/help"
-        DESTINATION "${INSTALL_DOC_DIR}")
+        "${CMAKE_SOURCE_DIR}/docs"
```
Comment (vomikan) [inline CMakeLists.txt]: http://desktop.moneymanagerex.org/ is a redirect of docs.

File: bootstrap.bat (hunk 1 above)
Comment (vomikan) [inline bootstrap.bat]: You are right.

File: update_cmake_build.bat (hunk 2 above)
Comment (vomikan) [inline update_cmake_build.bat]: I agree

File: update_cmake_build.bat (hunk 3 above)
Comment (vomikan) [inline update_cmake_build.bat]: TBD

File: CMakeLists.txt (hunk 7 above)
Comment (tomasfrtala) [inline CMakeLists.txt]: Ok, to the future I prepare another PR which takes into the consideration also test names in artifact naming convention.

**Comment by vomikan** [comment]
@stef145g what do you think about new vertion of the update_cmake_build_for_msv.bat file
https://github.com/moneymanagerex/moneymanagerex/pull/1136/commits/7d7f971762ab876835d85ed837f5520f0991774d ?

**Comment by stef145g** [comment]
I have checked the new file in ./util and while I like the fact that it produces a windowed result, the new file does not put the files in the correct place, therefore can not be used.

To change the location of _cmake_build_location_, in the .bat file, the CMakeLists.txt script must change the _EXECUTABLE_OUTPUT_PATH_ in line 113 to the same location. I don't know how this will affect the other builds.

The fact that you have restored the original .bat file, it can be **Rebased and Merged** into master.

**Comment by stef145g** [comment]
@vomikan just checked the original .bat file.
1. Where are the help files?
2. Are the replaced help files the same as the one's in master?

**Comment by stef145g** [comment]
The file: _update_cmake_build.bat_ has been modified to work with the changes in this PR in accordance to the original changes made to this file.

The file: _update_msvc_build_using_cmake.bat_ has been added to the **./util** with associated changes for it's relocation and renaming. 

These files can be found in: https://github.com/stef145g/moneymanagerex/tree/vomikan_master

**Pull request merged**